charts/nginz/templates/deployment.yaml

```diff
@@ -62,14 +62,14 @@ spec:
           httpGet:
             path: /status
             port: {{ .Values.config.http.httpPort }}
-            scheme: HTTP
+            scheme: {{ if .Values.config.http2.enabled }}HTTP2{{ else }}HTTP{{ end }}
         livenessProbe:
           initialDelaySeconds: 30
           timeoutSeconds: 1
           httpGet:
             path: /status
             port: {{ .Values.config.http.httpPort }}
-            scheme: HTTP
+            scheme: {{ if .Values.config.http2.enabled }}HTTP2{{ else }}HTTP{{ end }}
```

Comment:
http2 is not a scheme, that's either http or https (see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#httpgetaction-v1-core). It's up to the client to upgrade to http2 if it can.

In k8s, it probably already automatically upgrades if it can (see https://pkg.go.dev/golang.org/x/net/http2):

> Package http2 implements the HTTP/2 protocol.
> This package is low-level and intended to be used directly by very few people. Most users will use it indirectly through the automatic use by the net/http package (from Go 1.6 and later). For use in earlier Go versions see ConfigureServer. (Transport support requires Go 1.6 or later)

Reply:
Thanks! I saw that in some GCP thing where they explicitly put http2, but I should have checked the official docs.